backend/apps/main/models.py

```diff
@@ -1,68 +1,229 @@
+    rc_person = models.CharField(blank=True, max_length=MAX_LENGTH)
+    text = models.TextField(blank=True)
+    tid = models.CharField(unique=True, max_length=MAX_LENGTH)
+    title = models.CharField(blank=True, max_length=MAX_LENGTH)
+
+    authors = models.ManyToManyField(DjangoPerson, related_name="document_by_authors")
+    recipients = models.ManyToManyField(DjangoPerson, related_name="document_by_recipients")
+
+    def __str__(self):
+        return f'tid: {self.tid}, title: {self.title}, date: {self.date}'
 
 
-class Edge:
+def import_csv_to_document_model(csv_path):
     """
-    Python object to store edges, should be replaced if the Person class is replaced
+    Reads csv of docs and create Document model
+    :param csv_path: Path to csv file
+    :return: None
     """
-    def __init__(self, pk: int, node1: str, node2: str,  # pylint: disable-msg=R0913
-                 docs: int, words: int):
-        self.pk = pk  # pylint: disable=C0103
-        self.node1 = node1  # could be replaced with Person
-        self.node2 = node2
-        self.docs = docs
-        self.words = words
+    # Read csv into dataframe
+    docs = pd.read_csv(csv_path).fillna('')
+    # For each row, create & save the appropriate Document object
+    for _, row in docs.iterrows():
+        doc = Document(au=row['au'],
+                       au_org=row['au_org'],
+                       au_person=row['au_person'],
+                       cc=row['cc'],
+                       cc_org=row['cc_org'],
+                       collection=row['collection'],
+                       date=row['date'],
+                       doc_type=row['doc_type'],
+                       pages=int(row['pages']),
+                       rc=row['rc'],
+                       rc_org=row['rc_org'],
+                       rc_person=row['rc_person'],
+                       text=row['text'],
+                       tid=row['tid'],
+                       title=row['title'])
+        doc.save()
 
+        # for au/au_person, and rc/rc_person, parse it into list of individual raw names
+        # assumes that names are either in 'au_person'/'rc_person or 'au'/'rc', but not both (
+        # this is mostly true)
+        # (if 'au_person' is not empty, then it only parses info from 'au_person'; otherwise,
+        # parses 'au'; usually 'au_person' has more reliable information, 'au' may have erroneous
+        # info. Same for rc)
+        parsed_au = []
+        if row['au_person']:
+            parsed_au = parse_column_person(row['au_person'])
+        elif row['au']:
+            parsed_au = parse_column_person(row['au'])
 
-def load_network_json_data(return_type: str):
+        parsed_rc = []
+        if row['rc_person']:
+            parsed_rc = parse_column_person(row['rc_person'])
+        elif row['rc']:
+            parsed_rc = parse_column_person(row['rc'])
+
+        def match_djangoperson_from_name(parsed_name):
```

Comment:
I don't think this function needs to be a sub-function of import_csv. It's probably better to split it off into a separate function.